evap/evaluation/tools.py

```diff
@@ -177,21 +179,25 @@ def _calculate_results_impl(course):
                 answers = get_answers_from_answer_counters(answer_counters)
 
                 total_count = len(answers)
-                average = avg(answers) if total_count > 0 else None
-                deviation = pstdev(answers, average) if total_count > 0 else None
-                counts = get_counts(answer_counters)
-                warning = total_count > 0 and total_count < questionnaire_warning_thresholds[questionnaire]
+                if not export or total_count > 0:
+                    average = avg(answers) if total_count > 0 else None
+                    deviation = pstdev(answers, average) if total_count > 0 else None
+                    counts = get_counts(answer_counters)
+                    warning = total_count > 0 and total_count < questionnaire_warning_thresholds[questionnaire]
 
-                results.append(RatingResult(question, total_count, average, deviation, counts, warning))
+                    results.append(RatingResult(question, total_count, average, deviation, counts, warning))
 
             elif question.is_text_question:
+                if export:
+                    continue
                 allowed_states = [TextAnswer.PRIVATE, TextAnswer.PUBLISHED]
                 answers = get_textanswers(contribution, question, allowed_states)
                 results.append(TextResult(question=question, answers=answers))
 
         section_warning = questionnaire_max_answers[(questionnaire, contribution)] < questionnaire_warning_thresholds[questionnaire]
 
-        sections.append(ResultSection(questionnaire, contribution.contributor, contribution.label, results, section_warning))
+        if not export or results:
+            sections.append(ResultSection(questionnaire, contribution.contributor, contribution.label, results, section_warning))
```

Comment:
i would prefer skipping text questions and questions with no answers in the exporter. i'm wondering whether there are any code changes required at all? text questions seem to be skipped already.